doctests/home-json.js

```diff
@@ -0,0 +1,149 @@
+// EXAMPLE: js_home_json
+// REMOVE_START
+import assert from 'assert';
+// REMOVE_END
+// STEP_START import
+import {
+    createClient,
+    SchemaFieldTypes,
+    AggregateGroupByReducers,
+    AggregateSteps,
+} from 'redis';
+// STEP_END
+
+// STEP_START connect
+const client = await createClient();
+await client.connect();
+// STEP_END
+// REMOVE_START
+try {
+    await client.ft.dropIndex('idx:users', { DD: true });
+} catch{}
+
+await client.del('user:1', 'user:2', 'user:3');
+// REMOVE_END
+
+// STEP_START create_data
+var user1 = {
+    name: "Paul John",
+    email: "paul.john@example.com",
+    age: 42,
+    city: "London"
+}
+
+var user2 = {
+    name: "Eden Zamir",
+    email: "eden.zamir@example.com",
+    age: 29,
+    city: "Tel Aviv"
+}
+
+var user3 = {
+    name: "Paul Zamir",
+    email: "paul.zamir@example.com",
+    age: 35,
+    city: "Tel Aviv"
+}
+// STEP_END
+
+// STEP_START make_index
+await client.ft.create('idx:users', {
+    '$.name': {
+        type: SchemaFieldTypes.TEXT,
+        AS: 'name'
+    },
+    "$.city": {
+        type: SchemaFieldTypes.TEXT,
+        AS: 'city'
+    },
+    '$.age': {
+        type: SchemaFieldTypes.NUMERIC,
+        AS: 'age'
+    }
+}, {
+    ON: 'JSON',
+    PREFIX: 'user:'
+});
+// STEP_END
+
+// STEP_START add_data
+var user1Added = await client.json.set('user:1', '$', user1);
+var user2Added = await client.json.set('user:2', '$', user2);
+var user3Added = await client.json.set('user:3', '$', user3);
+// STEP_END
+// REMOVE_START
+assert.equal('OK', user1Added);
+assert.equal('OK', user2Added);
+assert.equal('OK', user3Added);
+// REMOVE_END
+
+// STEP_START query1
+var findPaulResult = await client.ft.search('idx:users', 'Paul @age:[30 40]');
+
+console.log(findPaulResult.total); // >>> 1
+
+findPaulResult.documents.forEach(doc => {
+    console.log(`ID: ${doc.id}, name: ${doc.value.name}, age: ${doc.value.age}`);
+});
+// >>> ID: user:3, name: Paul Zamir, age: 35
+// STEP_END
+// REMOVE_START
+assert.strictEqual(1, findPaulResult.total);
+
+var paulDoc = findPaulResult.documents[0];
+
+assert.equal('user:3', paulDoc.id);
+// REMOVE_END
+
+// STEP_START query2
+var citiesResult = await client.ft.search('idx:users', '*',{
+    RETURN: "city"
+});
+
+console.log(citiesResult.total); // >>> 3
+
+citiesResult.documents.forEach(cityDoc => {
+    console.log(cityDoc.value);
+});
+// >>> {city: 'London'}
+// >>> {city: 'Tel Aviv'}
+// >>> {city: 'Tel Aviv'}
+// STEP_END
+// REMOVE_START
+assert.strictEqual(3, citiesResult.total);
+
+citiesResult.documents.sort((a, b)=> a.id < b.id);
```

Comment:
why are we sorting on the client side and not using RediSearch it self?!

Reply:
This is just for the test code. For other client examples I've written, the order of the documents returned is not deterministic, so you need to sort before using the asserts, otherwise you get an occasionally flaky test. The reason I don't add the sort option to the command is simply that the other client examples don't use the sort option here - just keeping them all consistent.